third_party/forked/celopenapi/model/schemas.go

```diff
@@ -0,0 +1,398 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"k8s.io/kube-openapi/pkg/validation/spec"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+// SchemaDeclTypes constructs a top-down set of DeclType instances whose name is derived from the root
+// type name provided on the call, if not set to a custom type.
+func SchemaDeclTypes(s *spec.Schema, maybeRootType string) (*DeclType, map[string]*DeclType) {
+	root := SchemaDeclType(s).MaybeAssignTypeName(maybeRootType)
+	types := FieldTypeMap(maybeRootType, root)
+	return root, types
+}
+
+// SchemaDeclType returns the CEL Policy Templates type name associated with the schema element.
+func SchemaDeclType(s *spec.Schema) *DeclType {
+	if len(s.Type) > 1 {
+		// TODO
```

Comment:
I don't know the background on the decision to allow more than 1 type. https://github.com/kubernetes/kube-openapi/blob/b3fe75cc9b2fe5d5dc7929832b1516bb9b22362f/pkg/validation/validate/schema.go#L136 shows how the validator code in kube-openapi deals with the cardinality.

I've restructured this function slightly so that it tries all the types. I'm not 100% certain this is needed, but it should be safe.

Reply:
In that case, you'll have to return a `decls.Dyn` value for now and refine the experience later.